integration_tests/src/contract_test.rs

```diff
@@ -0,0 +1,200 @@
+            .to_address(),
+        get_altheadb_contract_addr(),
+    )
+    .await
+    .unwrap();
+
+    assert!(res.is_empty());
+
+    let res = get_registered_client_using_wgkey(
+        user.wg_public_key,
+        ETH_MINER_KEY
+            .parse::<clarity::PrivateKey>()
+            .unwrap()
+            .to_address(),
+        get_altheadb_contract_addr(),
+        &contact,
+    )
+    .await;
+
+    assert!(res.is_err());
+
+    // Add the first user
+    let _res = add_client_to_registered_list(
+        &contact,
+        user,
+        get_altheadb_contract_addr(),
+        ETH_MINER_KEY.parse().unwrap(),
+        None,
+        vec![],
+    )
+    .await
+    .unwrap();
+
+    thread::sleep(Duration::from_secs(5));
+
+    // Try requesting some info that doesnt exist
+    let res = get_registered_client_using_ethkey(
+        "0x3d261902a988d94599d7f0Bd4c2e4514D73BB329"
+            .parse()
+            .unwrap(),
+        ETH_MINER_KEY
+            .parse::<clarity::PrivateKey>()
+            .unwrap()
+            .to_address(),
+        get_altheadb_contract_addr(),
+        &contact,
+    )
+    .await;
+
+    assert!(res.is_err());
+
+    // Add the second user
+    let _res = add_client_to_registered_list(
+        &contact,
+        user_2,
+        get_altheadb_contract_addr(),
+        ETH_MINER_KEY.parse().unwrap(),
+        None,
+        vec![],
+    )
+    .await
+    .unwrap();
+
+    thread::sleep(Duration::from_secs(5));
+
+    // Add the third user
+    let _res = add_client_to_registered_list(
+        &contact,
+        user_3,
+        get_altheadb_contract_addr(),
+        ETH_MINER_KEY.parse().unwrap(),
+        None,
+        vec![],
+    )
+    .await
+    .unwrap();
+
+    thread::sleep(Duration::from_secs(10));
+
+    let res = get_all_regsitered_clients(
+        &contact,
+        ETH_MINER_KEY
+            .parse::<clarity::PrivateKey>()
+            .unwrap()
+            .to_address(),
+        get_altheadb_contract_addr(),
+    )
+    .await;
+
+    info!("All users are : {:?}", res);
+
+    thread::sleep(Duration::from_secs(5));
+
+    let res = get_registered_client_using_wgkey(
+        user.wg_public_key,
+        ETH_MINER_KEY
+            .parse::<clarity::PrivateKey>()
+            .unwrap()
+            .to_address(),
+        get_altheadb_contract_addr(),
+        &contact,
+    )
+    .await
+    .unwrap();
+    assert_eq!(res, user);
+
+    let res = get_registered_client_using_ethkey(
+        user_2.eth_address,
+        ETH_MINER_KEY
+            .parse::<clarity::PrivateKey>()
+            .unwrap()
+            .to_address(),
+        get_altheadb_contract_addr(),
+        &contact,
+    )
+    .await
+    .unwrap();
+    assert_eq!(res, user_2);
+
+    let res = get_registered_client_using_meship(
+        user_3.mesh_ip,
+        ETH_MINER_KEY
+            .parse::<clarity::PrivateKey>()
+            .unwrap()
+            .to_address(),
+        get_altheadb_contract_addr(),
```

Comment:
👍🏻  this is good as far as what's tested.

You could conserve a lot of lines by grabbing the ETH_MINER_KEY once at the top of the function and assigning it to a local var.

There's also a function in the debts tests in rita common that will generate random identities for you, might be useful to move that around so you can import it here and generate the identities, it's a little better at finding problems than hardcoded ID's and also is more compact.

Reply:
done